### ISSUE TITLE
Incomplete pat-autosuggest implementation

### ISSUE THREAD

**Issue by gyst** [ploneintranet/ploneintranet#317]
While working on personalized-tagging I stumble on the problem that setting tags is borked:
- Tags are stored as comma-joined ("foo,bar") instead of properly separated ("foo", "bar"). This is likely something that may have to be modified in ploneintranet.theme.utils.dexterity_update but I'm puzzled that the corresponding Mockup.select2 widget _does_ work properly.
- Tag suggestions are not properly hooked up via a AJAX lookup. I can see in the pat-autosuggest source that there appears to be support for this but have no idea how to properly hook this up.

Some related issues that may have to become separate tickets:
- ?diazo.off shows a borked themed page rather than a completely unstyled page
- switching javascript off results in an unusable incomplete edit form rather than a proper oldskool fallback.

Flagging this as prio:high since:
- Search faceting on tags won't work with these weird comma-joined tags
- If we ship this we will later have to split the tags and cleanup the tag storages (dublincore, network, catalog, solr)
- Tag suggestions in document editing will not work until the ajax integration is working
- Tag suggestions in microblog will contain the borked "foo,bar" tag options created by document saves

To reproduce: go to
http://localhost:8080/Plone/workspaces/open-market-committee/manage-information/public-bodies-reform
Click on the (i) icon to open the metadata panel
Inspect the tagging ("subjects") input

Compare this with the Barceloneta implementation screenshot below, taken from
http://cms.localhost:8080/Plone/workspaces/open-market-committee/manage-information/public-bodies-reform

![tagging](https://cloud.githubusercontent.com/assets/122351/8371632/7399472e-1bda-11e5-8461-4b9da6de55fa.png)

**Comment by cillianderoiste** [comment]
The saving of tags should be fixed in #330, but I didn't look at the other issues such as populating the suggestions.

**Comment by gyst** [comment]
OK would be great to get the AJAX hookup for @@getVocabulary?name='ploneintranet.network.vocabularies.Keywords in place. Should probably use the widget callflow as documented in network/behaviors/metadata.py to look that up, rather than hardcode.

Maybe @jcbrand can help out with the actual pattern args if he's back.

**Comment by cillianderoiste** [comment]
Thanks @gyst, I'll have a look today. 

I can't find network/behaviors/metadata.py. Perhaps it's in a branch? We usually need a custom AJAX view for pat-autosuggest anyway, but this sounds interesting.

**Comment by gyst** [comment]
Yes that's on branch personalized-tagging.

**Comment by cillianderoiste** [comment]
IIUC I need to customize @@getVocabulary for workspaces so that it returns a list of JSON objects `[{"text": "Rain", "id": "Rain"}, {"text": "Sun", "id": "Sun"}]` rather than an object with the list in "results": `{"total": 2, "results": [{"text": "Rain", "id": "Rain"}, {"text": "Sun", "id": "Sun"}]}`.

**Comment by gyst** [comment]
On 30/06/15 18:21, goibhniu wrote:

> IIUC I need to customize @@getVocabulary for workspaces so that it
> returns a list of JSON objects |[{"text": "Rain", "id": "Rain"},
> {"text": "Sun", "id": "Sun"}]| rather than an object with the list in
> "results": |{"total": 2, "results": [{"text": "Rain", "id": "Rain"},
> {"text": "Sun", "id": "Sun"}]}|.

I would implement that with a request arg e.g. &resultsonly=1

def **call**(self):  # on PersonalizedVocabularyView
  base = VocabularyView(self.context, self.request).get_vocabulary()
  if self.request.get('resultsonly', False):
    return base['results']
  return base

That is far easier than creating a custom view and getting the layer
precedence right.

Probably PersonalizedVocabularyView should be refactored to inherit from
VocabularyView directly rather than from BaseVocabularyView. But that's
another matter.

## :*CU#

```
Guido Stevens  |  +31.43.3618933  |  http://cosent.nl

s o c i a l   k n o w l e d g e   t e c h n o l o g y
```

**Comment by cillianderoiste** [comment]
Ah, nice! I'll do it like that.

**Comment by cillianderoiste** [comment]
@gyst I've opened a PR: #343 which hooks up the tag suggestions. There's a permission problem which prevents the suggestions from being returned, but I didn't investigate that further.

**Comment by gyst** [comment]
Fixed in #470
